src/shared/components/data-table-field.tsx

```diff
@@ -0,0 +1,196 @@
+import React, { useState } from "react";
+import { FieldProps } from "react-jsonschema-form";
+import css from "./data-table-field.module.scss";
+import { MockSensor } from "../../sensors/mock-sensor";
+import { AllCapabilities, Sensor, SensorCapabilityKey } from "../../sensors/sensor";
+import { SensorComponent } from "../../mobile-app/components/sensor";
+import { IVortexFormContext } from "./form";
+import { getURLParam } from "../utils/get-url-param";
+import { DeviceSensor } from "../../sensors/device-sensor";
+import { JSONSchema7 } from "json-schema";
+import { IFormUiSchema } from "../experiment-types";
+import { Icon } from "./icon";
+import { IUseSensorResult, useSensor } from "../../mobile-app/hooks/use-sensor";
+
+const useMockSensor = getURLParam("mockSensor") || false;
+
+let sensorInstance: Sensor | null = null;
+const getSensor = () => {
+  if (!sensorInstance) {
+    if (useMockSensor) {
+      sensorInstance = new MockSensor({
+        capabilities: AllCapabilities,
```

Comment:
I think you should create your own constant here with the current `AllCapabilities` values in case we add more in the future to  `AllCapabilities`.

Reply:
Is there anything wrong if we request all the capabilities but use just some of them later on? Are we concerned about power consumption?

Reply:
Its more about what devices it looks for in the BLE advertisement.  It takes the capabilities and then filters out all the devices that don't have those capabilities and then scans for the UUIDs for the capabilities on the filtered devices.  So imagine if we added a device that measured something like strain and then we added `strain` to the `AllCapabilities` list - the next time we compiled the sensor tags wouldn't be available as a connected device as they don't support strain.  I think the capabilities should be a property of the authored table schema eventually so that when authoring the user can select from a list of options.

Reply:
Thanks, I see. I've already changed that in this branch. We have a new property in the schema called `sensorFields`. I was using it to know which properties should take values from the sensor. So, I'll use the same array to generate the capabilities list.